synapse/rest/client/devices.py

```diff
@@ -283,19 +369,58 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         else:
             raise errors.NotFoundError("No dehydrated device available")
 
+    async def on_DELETE(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
```

Comment:
I think this endpoint will always be available under `/org.matrix.msc2697.v2/dehydrated_device` if `msc2697` is True, even though it's not defined in that MSC. That is probably fine but I just wanted to call it out!

Reply:
Hmm, my patch created a separate class to avoid exactly this scenario. @H-Shay did my patch not apply correctly with `git am` or why was his changed?

Reply:
> Hmm, my patch created a separate class to avoid exactly this scenario. @H-Shay did my patch not apply correctly with `git am` or why was his changed?

It appears that I incorrectly applied the patch - I believe this is fixed now.

Reply:
Hmm, `git am` still wasn't used, that's not directly problematic if you applied the patch correctly, but I rebased the patch onto your branch to make this easier. In the future you should be able to apply such patches by adding `.patch` the URL and running:

```bash
curl https://github.com/matrix-org/synapse/commit/777b3056531ff37c91014abbd029ac079d8d7c25.patch | git am
```

This will then preserve the diff and commit message as it was provided.

Reply:
@poljar - I applied the patch via `git apply`. That being said I think it applied correctly, are you okay with me going ahead and merging this?  I know there is a desire to get this in for the RC before tomorrow.